src/clients/detector.rs

```diff
@@ -47,39 +47,71 @@ impl DetectorClient {
     }
 }
 
-#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
-pub struct DetectorRequest {
-    pub text: String,
-    pub parameters: HashMap<String, serde_json::Value>,
+/// Request for text content analysis
+/// Results of this request will contain analysis / detection of each of the provided documents
+/// in the order they are present in the `contents` object.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ContentAnalysisRequest {
+    /// Field allowing users to provide list of documents for analysis
+    pub contents: Vec<String>,
 }
 
-impl DetectorRequest {
-    pub fn new(text: String, parameters: HashMap<String, serde_json::Value>) -> Self {
-        Self { text, parameters }
+impl ContentAnalysisRequest {
+    pub fn new(contents: Vec<String>) -> ContentAnalysisRequest {
+        ContentAnalysisRequest { contents }
     }
 }
 
-#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
-pub struct Detection {
+/// Evidence type
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EvidenceType {
+    Url,
+    Title,
+}
+
+/// Source of the evidence e.g. url
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Evidence {
+    /// Evidence source
+    pub source: String,
+}
+
+/// Evidence in response
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EvidenceObj {
+    /// Type field signifying the type of evidence provided
+    #[serde(rename = "type")]
+    pub r#type: EvidenceType,
+    /// Evidence currently only containing source
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub evidence: Option<Evidence>,
+}
+
+/// Response of text content analysis endpoint
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ContentAnalysisResponse {
+    /// Start index of detection
     pub start: usize,
+    /// End index of detection
     pub end: usize,
-    pub text: String,
+    /// Relevant detection class
     pub detection: String,
+    /// Detection type or aggregate detection label
     pub detection_type: String,
+    /// Score of detection
     pub score: f64,
+    /// Optional, any applicable evidences for detection
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub evidences: Option<Vec<EvidenceObj>>,
 }
 
-#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
-pub struct DetectorResponse {
-    pub detections: Vec<Detection>,
-}
-
-impl From<Detection> for crate::models::TokenClassificationResult {
-    fn from(value: Detection) -> Self {
+impl From<ContentAnalysisResponse> for crate::models::TokenClassificationResult {
+    fn from(value: ContentAnalysisResponse) -> Self {
         Self {
             start: value.start as u32,
             end: value.end as u32,
-            word: value.text,
+            word: "".to_string(), // Expected to be filled in
```

Comment:
nit: Can you add a TODO here that this value will be provided by the detectors API in next iteration so we won't have to do this for long?